File: erp-frontend/src/enums/biz/taskStatus.ts
import { BaseEnum, BaseEnumItem } from '@/enums/baseEnum';

/**
 * 任务状态枚举
 */
const TASK_STATUS: BaseEnum<string, string> = new BaseEnum<string, string>();
TASK_STATUS.set('WAIT_EVALUATION', new BaseEnumItem<string, string>('WAIT_EVALUATION', '待技术评估'));
TASK_STATUS.set('WAIT_DISPATCH', new BaseEnumItem<string, string>('WAIT_DISPATCH', '待派发'));
TASK_STATUS.set('EXECUTION', new BaseEnumItem<string, string>('EXECUTION', '维修执行'));
TASK_STATUS.set('RETURNED', new BaseEnumItem<string, string>('RETURNED', '任务退修'));
TASK_STATUS.set('CLOSED', new BaseEnumItem<string, string>('CLOSED', '任务关闭'));

export { TASK_STATUS };

File: erp-backend/xingyun-shkb/src/main/java/com/lframework/xingyun/shkb/enums/TaskStatus.java
package com.lframework.xingyun.shkb.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.lframework.starter.web.core.enums.BaseEnum;

/**
 * 合同任务状态
 */
    public enum TaskStatus implements BaseEnum<String> {
    WAIT_EVALUATION("WAIT_EVALUATION", "待技术评估"),
    WAIT_DISPATCH("WAIT_DISPATCH", "待派发"),
    EXECUTION("EXECUTION", "维修执行"),
    RETURNED("RETURNED", "任务退修"),
    // 任务关闭
    CLOSED("CLOSED", "任务关闭");



    @EnumValue
    private final String code;

    private final String desc;

    TaskStatus(String code, String desc) {

        this.code = code;
        this.desc = desc;
    }

    @Override
    public String getCode() {

        return this.code;
    }

    @Override
    public String getDesc() {

        return this.desc;
    }
}
